jerry-core/ecma/builtin-objects/ecma-builtin-bigint.c

```diff
@@ -36,6 +53,353 @@
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_bigint_object_as_int_n (ecma_value_t this_arg, /**< 'this' argument */
+                                     ecma_value_t bits, /**< number of bits */
+                                     ecma_value_t bigint, /**< bigint number */
+                                     bool is_signed) /**< The operation is signed */
+{
+  JERRY_UNUSED (this_arg);
+
+  ecma_number_t input_bits;
+  ecma_value_t bit_value = ecma_op_to_index (bits, &input_bits);
+
+  if (ECMA_IS_VALUE_ERROR (bit_value))
+  {
+    return bit_value;
+  }
+
+  ecma_value_t bigint_value = ecma_bigint_to_bigint (bigint, false);
+
+  if (ECMA_IS_VALUE_ERROR (bigint_value))
+  {
+    return bigint_value;
+  }
+
+  if (input_bits == 0 || bigint_value == ECMA_BIGINT_ZERO)
+  {
+    ecma_free_value (bigint_value);
+    return ECMA_BIGINT_ZERO;
+  }
+
+  const uint32_t size_of_divisor_in_bits = (sizeof (uint32_t) * sizeof (uint64_t));
+
+  uint64_t whole_part = (uint64_t) floor (input_bits / size_of_divisor_in_bits);
+  uint32_t remainder = (uint32_t) fmod (input_bits, size_of_divisor_in_bits);
+
+  uint64_t input_bit_length = 1;
+
+  if (whole_part == 0)
+  {
+    input_bit_length = (uint64_t) input_bits;
+  }
+  else
+  {
+    input_bit_length = (remainder > 0) ? (uint64_t) (whole_part + 1) * size_of_divisor_in_bits
+    : (uint64_t) (whole_part * size_of_divisor_in_bits);
+  }
+
+  uint64_t input_byte_size = (uint64_t) floor (input_bits / sizeof (uint64_t));
+
+  if (fmod (input_bits, sizeof (uint64_t)) != 0)
+  {
+    input_byte_size += 1;
+  }
+
+  ecma_extended_primitive_t *input_bigint_p = ecma_get_extended_primitive_from_value (bigint_value);
+
+  uint32_t bigint_size = ECMA_BIGINT_GET_SIZE (input_bigint_p);
+  uint8_t input_bigint_sign = input_bigint_p->u.bigint_sign_and_size & ECMA_BIGINT_SIGN;
+
+  const uint64_t input_bits_in_byte = (uint32_t) input_bit_length / sizeof (uint64_t);
+
+  uint64_t min_size = (input_bits_in_byte < bigint_size) ? input_bits_in_byte : bigint_size;
+
+  if (input_bigint_sign && (input_byte_size > bigint_size))
+  {
+    min_size = (input_bits_in_byte > bigint_size) ? input_bits_in_byte : bigint_size;
+  }
+
+  if (min_size < sizeof (uint32_t))
+  {
+    min_size = sizeof (uint32_t);
+  }
+
+  ecma_extended_primitive_t * result_p = ecma_bigint_create ((uint32_t) min_size);
+
+  if (JERRY_UNLIKELY (result_p == NULL))
+  {
+    ecma_deref_bigint (input_bigint_p);
+    return ECMA_VALUE_ERROR;
+  }
+
+  ecma_bigint_digit_t *first_digit_p = ECMA_BIGINT_GET_DIGITS (input_bigint_p, 0);
+  ecma_bigint_digit_t *last_digit_p = ECMA_BIGINT_GET_DIGITS (input_bigint_p, bigint_size);
+
+  /* Calculate the leading zeros of the input_bigint */
+  ecma_bigint_digit_t zeros;
+
+  do
+  {
+    zeros = ecma_big_uint_count_leading_zero (*first_digit_p);
+    ++first_digit_p;
+  }
+  while (first_digit_p < last_digit_p);
```

Comment:
What is the exact purpose of this while? Isn't it the same as `zeros = ecma_big_uint_count_leading_zero (last_digit_p[-1]);`